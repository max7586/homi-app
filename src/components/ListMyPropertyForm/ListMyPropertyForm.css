.ListMyPropertyForm__submit {
    margin-bottom: 70px;
}