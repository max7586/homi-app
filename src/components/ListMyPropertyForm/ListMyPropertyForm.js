import React, { Component } from 'react'
import { Button, Input, Required } from '../Utils/Utils'

import PropertyApiService from '../../services/property-api-service'

export default class ListMyPropertyForm extends Component {
  static defaultProps = {
    onAddPropertySuccess: () => {}
  }

  state = { error: null }

  handleSubmit = ev => {
    ev.preventDefault()
// alert("clicked")
    // title is an address 
    
    const { title, content, image} = ev.target

     this.setState({ error: null })
     PropertyApiService.postProperty(title.value, content.value, image.value) 
     .then(this.context.addProperty) 
     .then( property => {
      title.value = ''
      content.value = ''
      image.value = ''

      this.props.onAddPropertySuccess()
             })
         .catch(res => {
           this.setState({ error: res.error })
         })
    }


  render() {
    const { error } = this.state
    return (
      <form
        className='ListMyPropertyForm'
        onSubmit={this.handleSubmit}
      >
        <div role='alert'>
          {error && <p className='red'>{error}</p>}
        </div>
        <div className='title'>
          <label htmlFor='ListMyPropertyForm__title'>
            Address <Required />
          </label>
          <Input
            name='title'
            placeholder='101 street, city, state, ZIP code'
            type='text'
            required
            id='ListMyPropertyForm__title'>
          </Input>
        </div>

        <div className='image'>
          <label htmlFor='ListMyPropertyForm__image'>
            Image URL  <Required />
          </label>
          <Input
            name='image'
            placeholder= 'https://loremflickr.com/750/300/house?random'
            type='text'
            required
            id='ListMyPropertyForm__image'>
          </Input>
        </div>
        <div className='content'>
          <label htmlFor='ListMyPropertyForm__content'>
            Propertiy Description <Required />
          </label>
          <Input
            name='content'
            placeholder='Detail about house, 560 Sq-Ft 6 Bedrooms 2 Garage 5 Bathroom. FOR RENT'
            type='content'
            required
            id='ListMyPropertyForm__content'>
          </Input>
        </div>

        <Button type='submit'>
          Submit
        </Button>
      </form>
    )
  }
}
