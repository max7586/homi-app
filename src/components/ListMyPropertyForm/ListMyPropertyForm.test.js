import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ListMyPropertyForm from './ListMyPropertyForm';
import { BrowserRouter } from 'react-router-dom'

describe('ListMyPropertyForm component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <BrowserRouter>
        <ListMyPropertyForm />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ListMyPropertyForm onSuccess={()=>{}}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders the UI as expected with no props', () => {
    const tree = renderer
      .create(<ListMyPropertyForm />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});