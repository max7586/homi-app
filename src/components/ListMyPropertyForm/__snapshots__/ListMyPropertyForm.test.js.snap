// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListMyPropertyForm component renders the UI as expected 1`] = `
<form
  className="ListMyPropertyForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="title"
  >
    <label
      htmlFor="ListMyPropertyForm__title"
    >
      Address 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="ListMyPropertyForm__title"
      name="title"
      placeholder="101 street, city, state, ZIP code"
      required={true}
      type="text"
    />
  </div>
  <div
    className="image"
  >
    <label
      htmlFor="ListMyPropertyForm__image"
    >
      Image URL  
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="ListMyPropertyForm__image"
      name="image"
      placeholder="https://loremflickr.com/750/300/house?random"
      required={true}
      type="text"
    />
  </div>
  <div
    className="content"
  >
    <label
      htmlFor="ListMyPropertyForm__content"
    >
      Propertiy Description 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="ListMyPropertyForm__content"
      name="content"
      placeholder="Detail about house, 560 Sq-Ft 6 Bedrooms 2 Garage 5 Bathroom. FOR RENT"
      required={true}
      type="content"
    />
  </div>
  <button
    className="Button ListMyPropertyForm__submit"
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`ListMyPropertyForm component renders the UI as expected with no props 1`] = `
<form
  className="ListMyPropertyForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="title"
  >
    <label
      htmlFor="ListMyPropertyForm__title"
    >
      Address 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="ListMyPropertyForm__title"
      name="title"
      placeholder="101 street, city, state, ZIP code"
      required={true}
      type="text"
    />
  </div>
  <div
    className="image"
  >
    <label
      htmlFor="ListMyPropertyForm__image"
    >
      Image URL  
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="ListMyPropertyForm__image"
      name="image"
      placeholder="https://loremflickr.com/750/300/house?random"
      required={true}
      type="text"
    />
  </div>
  <div
    className="content"
  >
    <label
      htmlFor="ListMyPropertyForm__content"
    >
      Propertiy Description 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="ListMyPropertyForm__content"
      name="content"
      placeholder="Detail about house, 560 Sq-Ft 6 Bedrooms 2 Garage 5 Bathroom. FOR RENT"
      required={true}
      type="content"
    />
  </div>
  <button
    className="Button ListMyPropertyForm__submit"
    type="submit"
  >
    Submit
  </button>
</form>
`;
